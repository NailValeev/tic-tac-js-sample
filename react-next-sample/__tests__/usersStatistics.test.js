/* eslint-env jest */

import { shallow } from 'enzyme'
import React from 'react'
import renderer from 'react-test-renderer'

import UserStatistics from '../pages/usersStatistics.js'

describe('UserStatistics, with Enzyme', () => {
  it('UserStatistics shows header "User statistics: labor effectiveness"', () => {
    const props = { utils: [], total: [], pie: [], delta: [], errors : ['No errors, it is just a mock'] }
    var wrapper = shallow(<UserStatistics {...props} />)

      expect(wrapper.find('h2').text()).toEqual('User statistics: labor effectiveness')

  })
})

describe('UserStatistics, with Snapshot Testing', () => {
  it('App shows table header "User statistics: labor effectiveness"', () => {
    const props = { utils: [], total: [], pie: [], delta: [], errors : ['No errors, it is just a mock'] }
    const component = renderer.create(<UserStatistics {...props}/>)
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
