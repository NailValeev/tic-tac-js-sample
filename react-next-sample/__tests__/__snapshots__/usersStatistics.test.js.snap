// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserStatistics, with Snapshot Testing App shows table header "User statistics: labor effectiveness" 1`] = `
<div
  style={
    Object {
      "border": "1px solid #DDD",
      "margin": 20,
      "padding": 20,
    }
  }
>
  <div
    className="jsx-1928146253 blue"
  >
    <a
      className="jsx-1928146253 btn btn1"
      href="/"
      onClick={[Function]}
    >
      Home
    </a>
    <a
      className="jsx-1928146253 btn btn2"
      href="/userstats"
      onClick={[Function]}
    >
      Users statistics
    </a>
    <a
      className="jsx-1928146253 btn btn3"
      href="/projectstats"
      onClick={[Function]}
    >
      Projects statistics
    </a>
    <a
      className="jsx-1928146253 btn btn4"
      href="/plans"
      onClick={[Function]}
    >
      Planned time
    </a>
    <a
      className="jsx-1928146253 btn btn5"
      href="/trends"
      onClick={[Function]}
    >
      Trends
    </a>
  </div>
  <h2>
    User statistics: labor effectiveness
  </h2>
  <p>
    <b>
      Resource Utilization
    </b>
     = time spent on useful tasks / total amount of time available to work
  </p>
  <p>
    <b>
      Billable Utilization
    </b>
     = time spent on billable tasks / total amount of time available to work
  </p>
  <h3
    className="errorMsg"
  >
    No errors, it is just a mock
  </h3>
  <div>
    Loading Chart
  </div>
  <div>
    Loading Chart
  </div>
  <div>
    Loading Chart
  </div>
  <div>
    Loading Chart
  </div>
</div>
`;
